allennlp/semparse/executors/wikitables_variable_free_executor.py

```diff
@@ -0,0 +1,647 @@
+            return False
+        return self > other or self == other
+
+
+
+class WikiTablesVariableFreeExecutor(Executor):
+    def __init__(self, table_data: List[Dict[str, str]]) -> None:
+        super().__init__(table_data)
+        self._table_data = table_data
+
+    @overrides
+    def _handle_expression(self, expression_list: NestedList):
+        if isinstance(expression_list, list) and len(expression_list) == 1:
+            expression = expression_list[0]
+        else:
+            expression = expression_list
+        if isinstance(expression, list):
+            # This is a function application.
+            function_name = expression[0]
+        else:
+            # This is a constant (like "all_rows" or "2005")
+            return self._handle_constant(str(expression))
+        try:
+            function = getattr(self, f"_{function_name}")
+            return function(expression[1:])
+        except AttributeError:
+            logger.error("Function not found: %s", function_name)
+            raise ExecutionError(f"Function not found: {function_name}")
+
+    def _get_row_list_and_column_name(self, expression_list: NestedList) -> Tuple[List[Dict[str,
+                                                                                            str]],
+                                                                                  str]:
+        """
+        Utility function for computing the initial row list and a column name from an expression for
+        all functions that need these operations, like "select", "argmax", "argmin", etc.
+        """
+        row_list: List[Dict[str, str]] = self._handle_expression(expression_list[0])
+        column_name = expression_list[1]
+        if not (isinstance(column_name, str) and column_name.startswith("fb:row.row.")):
+            logger.error("Invalid column for selection: %s", column_name)
+            raise ExecutionError(f"Invalid column for selection: {column_name}")
+        if row_list and column_name not in row_list[0]:
+            logger.error("Input list of rows do not contain column: %s", column_name)
+            raise ExecutionError(f"Input list of rows do not contain column: {column_name}")
+        return row_list, column_name
+
+    def _handle_constant(self, constant: str) -> Union[List[Dict[str, str]], float]:
+        if constant == "all_rows":
+            return self._table_data
+        try:
+            return float(constant)
+        except ValueError:
+            logger.error(f"Cannot handle constant: {constant}")
+            raise ExecutionError(f"Cannot handle constant: {constant}")
+
+    def _select(self, expression_list: NestedList) -> List[str]:
+        """
+        Select function takes a list of rows and a column (decoded from the `expression_list`) and
+        returns a list of cell values as strings.
+        """
+        row_list, column_name = self._get_row_list_and_column_name(expression_list)
+        if not row_list:
```

Comment:
If `row_list` is an empty list, you'll get this behavior for free by just having the return line below.